frontend/spec/features/add_as_you_go_spec.rb

```diff
@@ -0,0 +1,379 @@
+      end
+
+      it "within the Contact Details subform for #{type} agents" do
+        contact_details_subform = page.find "#agent_#{type}_contact_details"
+        within contact_details_subform do
+          click_button 'Add Contact'
+          click_button 'Add Telephone Number'
+          click_button 'Add Contact Note'
+        end
+        expect(contact_details_subform).to have_css '[data-object-name="telephone"] > .subrecord-add-as-you-go-actions > a[title="Add Telephone Number"]', visible: false
+        expect(contact_details_subform).to have_css '#agent_contact_notes.notes-form > .subrecord-add-as-you-go-actions > a[title="Add Contact Note"]', visible: false
+        contact_details_subform.hover
+        expect(contact_details_subform).to have_css '[data-object-name="telephone"] > .subrecord-add-as-you-go-actions > a[title="Add Telephone Number"]', visible: true
+        expect(contact_details_subform).to have_css '#agent_contact_notes.notes-form > .subrecord-add-as-you-go-actions > a[title="Add Contact Note"]', visible: true
+      end
+
+      unless type == 'software'
+        it "within the Related External Resources subform for #{type} agents" do
+          related_agents_subform = page.find "#agent_#{type}_agent_resource"
+          within related_agents_subform do
+            click_button 'Add Related External Resource'
+            click_button 'Add Place'
+            click_button 'Add Date'
+          end
+          expect(related_agents_subform).to have_css '[data-object-name="place"] > .subrecord-add-as-you-go-actions > a[title="Add Place"]', visible: false
+          expect(related_agents_subform).to have_css '[data-object-name="date"] > .subrecord-add-as-you-go-actions > a[title="Add Date"]', visible: false
+          related_agents_subform.hover
+          expect(related_agents_subform).to have_css '[data-object-name="place"] > .subrecord-add-as-you-go-actions > a[title="Add Place"]', visible: true
+          expect(related_agents_subform).to have_css '[data-object-name="date"] > .subrecord-add-as-you-go-actions > a[title="Add Date"]', visible: true
+        end
+      end
+    end
+
+    it_behaves_like 'top-level non-Notes subforms', 'person', 20, 20
+    it_behaves_like 'top-level non-Notes subforms', 'family', 19, 19
+    it_behaves_like 'top-level non-Notes subforms', 'corporate_entity', 19, 19
+    it_behaves_like 'top-level non-Notes subforms', 'software', 10, 10
+
+    it_behaves_like 'top-level Notes subform', 'person'
+    it_behaves_like 'top-level Notes subform', 'family'
+    it_behaves_like 'top-level Notes subform', 'corporate_entity'
+    it_behaves_like 'top-level Notes subform', 'software'
+
+    it_behaves_like 'nested subforms', 'person'
+    it_behaves_like 'nested subforms', 'family'
+    it_behaves_like 'nested subforms', 'corporate_entity'
+    it_behaves_like 'nested subforms', 'software'
+  end
+
+  it 'are not present for Container Profiles' do
```

Comment:
How about adding a `context` block and making the `it` block more descriptive, something like:
```
context "when creating Container Profiles" do
  it "does not contain notes subforms" do
    ...
```